packages/cloud_functions/cloud_functions/lib/src/firebase_functions.dart

```diff
@@ -73,7 +73,7 @@ class FirebaseFunctions extends FirebasePluginPlatform {
     assert(origin.isNotEmpty);
 
     // Android considers localhost as 10.0.2.2 - automatically handle this for users.
-    if (defaultTargetPlatform == TargetPlatform.android) {
+    if (!kIsWeb && defaultTargetPlatform == TargetPlatform.android) {
```

Comment:
What about an assert?

Like:

```dart
assert(
  kIsWeb && defaultTargetPlatform != TargetPlatform.web, 
  '''
Tried to use `useFunctionsEmultator` on web, but `defaultTargetPlatform` indicated that the platform is android`.

This can happen inside tests when <add explanation>.
To fix, consider using `debugDefaultTargetPlatformOverride` like so:

void main() {
  debugDefaultTargetPlatformOverride = TargetPlatform.web;
  test('some test', () {...})
}
```

Reply:
@rrousselGit No `TargetPlatform.web` in stable :(

Reply:
Hum that's a shame, I don't mind then!

Reply:
@rrousselGit The only way to indicate whether it is Web or not - `kIsWeb`, and currently there is not `TargetPlatform.web` in Flutter (stable channel)